ext/storage_sources/azure_store/azure_connection.cpp

```diff
@@ -95,3 +102,50 @@ azure_connection::get_object(const std::string &path) const
 {
     return 0;
 }
+
+int
+azure_connection::object_exists(const std::string &object_key, bool &exists) const
+{
+    exists = false;
+    std::string obj = _object_prefix + object_key;
+
+    auto list_blob_response = _azure_client.ListBlobs();
+
+    for (const auto blob_item : list_blob_response.Blobs) {
+        // Check if object exists.
+        if (blob_item.Name.compare(obj) == 0) {
+            exists = true;
+            // Check if object is deleted and has not been cleared by garbage collection.
+            if (blob_item.IsDeleted) {
+                exists = false;
+                return -1;
+            }
+        }
+    }
+    return 0;
+}
+
+int
+azure_connection::bucket_exists(bool &exists) const
+{
+    exists = false;
+
+    auto service_client = Azure::Storage::Blobs::BlobServiceClient::CreateFromConnectionString(
+      std::getenv("AZURE_STORAGE_CONNECTION_STRING"));
+
+    // Get list of containers associated with the class Azure client.
+    auto list_container_response = service_client.ListBlobContainers();
+
+    for (const auto container_item : list_container_response.BlobContainers) {
+        // Check if bucket exists.
+        if (container_item.Name.compare(_bucket_name) == 0) {
+            exists = true;
```

Comment:
It's a little nitpicky, but can you move the `exists = true` down below the `IsDeleted` check? That way the returned value isn't "wrong" for a brief period (and you can get rid of the second `exists = false`). This goes for both `bucket_exists` and `object_exists`.

Reply:
that's a really good point. Makes a lot of sense. Done :)